packages/members-api/lib/stripe/index.js

```diff
@@ -178,6 +182,19 @@ module.exports = class StripePaymentProcessor {
         });
     }
 
+    /**
+     * Handles immediate subscription deletion
+     * @param {Object} subscription
+     */
+    async handleCustomerSubscriptionDelete(subscription) {
+        return await this.storage.set({
```

Comment:
@allouis have done the cancellation initiated by the user in this way, so that it doesn't interfere with the webhook's logic. Think it can be refactored into something nicer once the subscription model is passed in directly. Wdyt?

Reply:
Hmmm, I would've thought we'd want to update the `cancelSubscription` method to be like
```js
const cancelledSubscription = await del(this._stripe, blah blah); // this returns full subscription obj
await this._updateSubscription(cancelledSubscription);
return cancelledSubscription;
```

And use that method everywhere instead of raw calls to `del(.....` :thinking: